MailCore/Models/Draft.swift

```diff
@@ -482,7 +185,120 @@ public class Draft: Object, Decodable, Identifiable, AbstractDraft {
         self.action = action
     }
 
-    public func asUnmanaged() -> UnmanagedDraft {
-        return .toUnmanaged(managedDraft: self)
+    public static func mailTo(subject: String?,
+                              body: String?,
+                              to: [Recipient], cc: [Recipient], bcc: [Recipient]) -> Draft {
+        return Draft(subject: subject ?? "",
+                     body: body ?? "",
+                     to: to,
+                     cc: cc,
+                     bcc: bcc)
+    }
+
+    public static func writing(to recipient: Recipient) -> Draft {
+        return Draft(to: [recipient.detached()])
+    }
+
+    public static func replying(to message: Message, mode: ReplyMode, localDraftUUID: String) -> Draft {
+        var subject = "\(message.formattedSubject)"
+        let quote: String
+        var attachments: [Attachment] = []
+        switch mode {
+        case .reply, .replyAll:
+            if !subject.starts(with: "Re: ") {
+                subject = "Re: \(subject)"
+            }
+            quote = Constants.replyQuote(message: message)
+        case let .forward(attachmentsToForward):
+            if !subject.starts(with: "Fwd: ") {
+                subject = "Fwd: \(subject)"
+            }
+            quote = Constants.forwardQuote(message: message)
+            attachments = attachmentsToForward
+        }
+
+        var to: [Recipient] = []
+        var cc: [Recipient] = []
+
+        if mode.isReply {
```

Comment:
If we start writing some more unit tests, in the future keeping a dedicated func for the isReply case could be usefull.
That would mean this replying(to:) func would only be here to dispatch the correct specialized func.

Reply:
discussed